### DISCUSSION TITLE
dissocPath

### DISCUSSION THREAD

**Discussion by fneco** [remeda/remeda#961] [Suggest a Function]
This issue proposes the addition of a new function to the remeda library.

## Proposal

I would like to add a function that can delete data by specifying a path(=array of props).
The name `dissocPath`  is tentative.

| rEmEda      | array    | object   | array and object (with array of props) |
| ----------- | -------- | -------- | -------------------------------------- |
| read data   | `prop` | `prop` | `pathOr`                             |
| set data    | \-       | `set`  | -                      |
| delete data | \-       | \-       |  **`dissocPath`**                                    |
| update data | \-       | `set`  | `setPath`         |

## Motivation

As noted in the following table, there is no function in remeda to delete data by specifying a path(=array of props).

### As-Is RAMDA implementation for read/set/delete/update data by specifying property

| rAmda       | array                   | object     | array and object (with array of props) |
| ----------- | ----------------------- | ---------- | -------------------------------------- |
| read data   | `nth` / `prop`      | `prop`   | `path`                               |
| set data    | `insert` / `assoc`  | `assoc`  | `assocPath`                          |
| delete data | `remove` / `dissoc` | `dissoc` | `dissocPath`                         |
| update data | `update` / `assoc`  | `assoc`  | `assocPath`                          |

version: v0.29.1

### As-Is remeda implementation for read/set/delete/update data by specifying property

| rEmEda      | array    | object   | array and object (with array of props)             |
| ----------- | -------- | -------- | -------------------------------------------------- |
| read data   | `prop` | `prop` | `pathOr`                                         |
| set data    | \-       | `set`  | \-  |
| delete data | \-       | `omit` can do the same    |\-                                                   |
| update data | \-       | `set`  | `setPath`                                        |

commit hash: 78075dc

**Comment by eranhirsch** [comment]
I'm not a huge fan of the "path" based APIs, they are hard to type correctly so they don't provide the kind of type-safety I personally strive for for this library (compile-time errors are better than runtime-errors). But I agree that we don't provide a lot of ergonomics for working with deeply-nested objects which are common when processing responses from API servers.

@TkDodo, wdyt?

**Comment by TkDodo** [comment]
I'm fine with adding it
